src/diagram/components/ui/expandable-input.tsx

```diff
@@ -30,6 +30,14 @@ export const ExpandableInput = ({
     return val && val.toString().length >= length;
   };
 
+  const isValidValue = () => {
+    // Don't output NaN as a computed value.
+    if (value === "NaN" && placeholder === "value" && disabled) {
```

Comment:
Thanks. I was (somewhat) incorrectly using that placeholder check to ensure only the computed value field was altered, not the unit field. Teale's suggested change uses the more correct `inputType` which I misunderstood until now. It may be silly, but I was thinking we should keep this specific to the computed value field and leave the unit field alone...even though no one should ever enter NaN as a unit.

Reply:
I think we should disable `NaN` for all number input fields. But if we did want some to show `NaN` (for example, the unit field), I think it would make more sense to add a new optional `allowNaN` prop. That way, the client can decide whether they want to allow it or not, and we wouldn't have to explicitly list which fields do or don't use it in the component.